mobile/src/notifications/notification-listener.ts

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+ * Copyright (c) Arcadia, Inc. All rights reserved.
+ ******************************************************************************/
+
+import { ActionsObservable, combineEpics, ofType, StateObservable } from 'redux-observable';
+import { AuthActionType, UserLoggedIn } from '../reducers/auth/auth.action';
+import { catchError, ignoreElements, map, switchMap } from 'rxjs/operators';
+import Push from 'appcenter-push';
+import { EMPTY, from, of, Subject } from 'rxjs';
+import { Action } from 'redux';
+import { openProfile } from '../navigation/navigation.actions';
+import { AppState, DependenciesContainer } from '../reducers/app.reducer';
+import AppCenter from 'appcenter';
+import { installIdReceived, NotificationAction, NotificationActionType } from './notification.actions';
+
+//----------------------------------------------------------------------------
+const notificationsHandler$ = (action$: ActionsObservable<UserLoggedIn>) =>
+    action$.ofType(AuthActionType.userLoggedIn).pipe(
+        switchMap(() => {
+            const notification$ = new Subject<Action>();
+
+            Push.setListener({
+                onPushNotificationReceived: notification => {
+                    if (!notification.customProperties) {
+                        return;
+                    }
+
+                    notification$.next(openProfile());
+                },
+            });
+
+            return notification$;
+        }),
+    );
+
+//----------------------------------------------------------------------------
+const getInstallId$ = (action$: ActionsObservable<UserLoggedIn>) => action$.pipe(
+    ofType(AuthActionType.userLoggedIn),
+    switchMap(() => {
+        return from(AppCenter.getInstallId());
+    }),
+    map(installId => installIdReceived(installId)),
+);
+
+//----------------------------------------------------------------------------
+const notificationsRegister$ = (action$: ActionsObservable<NotificationAction>, _: StateObservable<AppState>, deps: DependenciesContainer) => action$.pipe(
+    ofType(NotificationActionType.installIdReceived),
+    switchMap(action => {
+        return deps.apiClient.put(`/push/device/${action.installId}`);
+    }),
+    catchError(error => {
+        console.warn(error);
+        return EMPTY;
```

Comment:
I believe user should be warned that he won't get push notifications in this case.

Reply:
An appropriate alert with only one "Ok" button is enough IMHO, but, of course, it's not a necessary change.

Reply:
And what be a proper message in this case? Something like "Application wasn't able to register for Push Notifications on the server, Push Notifications will not be received." or something?

Reply:
Yes, or like "An error occurred ...error description". Push Notifications will not be received. Please contact your service administrators."

Push notifications are insidious, so I want to prevent getting bugs with description like "Push notifications are not received in some cases. Please investigate".

Reply:
Well, makes sense.